File: client/src/models/Link.ts
/**
 * Link class
 *
 * @author Fabien Bellanger
 */
export default class Link {
    id = '';
    url = '';
    name = '';
    expired_at = '';
    created_at = '';

    constructor(
        id: string,
        url: string,
        name: string,
        expired_at: string,
        created_at: string
    ) {
        this.id = id;
        this.url = url;
        this.name = name;
        this.expired_at = (expired_at.length > 10) ? expired_at.substring(0, 10) : expired_at;
        this.created_at = (created_at.length > 10) ? created_at.substring(0, 10) : created_at;
    }

    isExpired(): boolean {
        return false;
    }
}

File: server/models/link.go
package models

import (
	"time"

	"gorm.io/gorm"
)

// Link represents a shortened URL.
type Link struct {
	ID        string         `json:"id" xml:"id" form:"id" gorm:"primaryKey;size:8" validate:"required"`
	URL       string         `json:"url" xml:"url" form:"url" gorm:"index;size:191" validate:"required"`
	Name      *string        `json:"name" xml:"name" form:"name" gorm:"index;size:127"`
	CreatedAt time.Time      `json:"created_at" xml:"created_at" form:"created_at" gorm:"autoCreateTime"`
	ExpiredAt time.Time      `json:"expired_at" xml:"expired_at" form:"expired_at"`
	DeletedAt gorm.DeletedAt `json:"-" xml:"-" form:"deleted_at" gorm:"index"`
}

// LinkForm is used to create or update a link.
type LinkForm struct {
	URL       string    `json:"url" xml:"url" form:"url" validate:"required"`
	Name      *string   `json:"name" xml:"name" form:"name"`
	ExpiredAt time.Time `json:"expired_at" xml:"expired_at" form:"expired_at"`
}
